### PR TITLE
WIP blogpost and complete-ish architecture draft

### PR DESCRIPTION
**Pull request by dlwh**

### CODE REVIEW

File: docs/Announcement-Post.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+* TPU support (naturally)
+* Flexible tensor parallelism support
+* Bitwise reproducibility
+* Improved support for resuming from preemption
+* (Coming Soon) Simultaneous preprocessing and training
+
+Levanter is not yet at complete feature parity with Mistral. Beyond the obvious lack of PyTorch support, Levanter does
+not yet support multi-machine training with GPUs, due to some issues with CUDA configuration we haven't quite worked out
+yet. Some libraries (e.g. WandB) have better PyTorch integration than Jax integration, so some features aren't quite
```
Comment (Ivan-Zhou) [inline docs/Announcement-Post.md:112]: Maybe we can link the GH issue here.

File: docs/Announcement-Post.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+you specify. We then use `hax.axis_mapping` to change the preferred mapping for computing the loss and gradients. Jax
+handles most of the heavy lifting, except Linear layers need a bit of handholding, which Haliax provides.
+
+Tensor Parallelism works in a broadly similar manner, with you specifying how "wide" you want model parallelism to be
+and specifying which axes of your model should be sharded. (For a transformer, this would typically be the Head
+and MLP axes.)
+
+In our experiments, we have found that, for models of the sizes that we can reasonably train with our resources, FSDP
+is more than enough to get good scaling, especially on TPU. In practice, with FSDP, we haven't found tensor parallelism
+to be useful for training transformers to at least 20B parameters. (Tensor parallelism is useful for inference, but
```
Comment (Ivan-Zhou) [inline docs/Announcement-Post.md:205]: This sentence is bit hard to understand, but I don't have better idea to rewrite it.

So I asked chatgpt, here's its suggestion:
> Through our experiments, we discovered that Fully Sharded Data Parallel (FSDP) provides sufficient scaling for the model sizes we can feasibly train using our available resources, particularly on Tensor Processing Units (TPU). In practical applications, using FSDP, we observed that tensor parallelism offered limited utility in training transformers with up to 20 billion parameters.

File: docs/Announcement-Post.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+Levanter's focus is on training.)
```
Comment (Ivan-Zhou) [inline docs/Announcement-Post.md:206]: Curious how long will this hold true?

File: docs/Announcement-Post.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+For more details, please see the [Architectural Overview](XXX) in the Levanter documentation.
+
+#### Bitwise Reproducibility
+
+One of the benefits of Jax is that it offers strong guarantees for reproducibility. In particular, Jax's fine-grained
+control over random number generation makes it easy to ensure bitwise reproducibility, especially when using TPUs.
+Levanter takes advantage of this to offer bitwise reproducibility for training runs, even after preemption. As an example,
+here is a screenshot of a training run being resumed multiple times, even on different TPU pod slices:
+
+![plot showing bitwise reproducibility with four training runs superimposed with the exact same loss curve](figures/bitwise_repro_curve.png)
```
Comment (Ivan-Zhou) [inline docs/Announcement-Post.md:217]: I found the highlighted area on the top left bit distracting. I thought you want to highlight that area, but in reality I guess you are not: 
CleanShot 2023-05-08 at 19 48 13

File: docs/Announcement-Post.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+been a significant challenge. In particular, it can be difficult to identify issues in the data that are causing
+particularly low loss. As a small step towards addressing this, we have added a feature to Levanter that allows you
+to visualize the probability of each token in a sample of the validation set during training. For novel data sets,
+this has allowed us to identify, e.g., highly but not perfectly redundant data (what we call "madlib duplicates") or other
+issues that might be causing abnormally low loss. We've also used it to qualitative assess how alternative architectures
+learn differently from Transformers.
+
+Here is an example of the token probability visualization in action:
+
+XXX legal data?
```
Comment (Ivan-Zhou) [inline docs/Announcement-Post.md:234]: Feel free to use example from backpack if you need: https://wandb.ai/stanford-mercury/levanter/runs/eiv0evi3kv93483620128070.0002?workspace=user-ivanzhou
CleanShot 2023-05-08 at 19 50 44

File: docs/Announcement-Post.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+
+This visualization is logged to WandB as training progresses, which is a nice alternative to just staring obsessively at
+the loss curve, which is what I usually do.
```
Comment (Ivan-Zhou) [inline docs/Announcement-Post.md:237]: +1 :)

File: docs/Announcement-Post.md [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+## Released Models
+
+Along with the release of the code, we are releasing a few models trained using Levanter. These models are available on
+the [HF model hub](XXX) and can be used with the Hugging Face Transformers library. We have more in development and will
+be releasing them as they become available.
+
+XXX TODO: verify this is what we're releasing, add links
+- GPT-2-1536 (1.45B parameters) XXX, including checkpoints every 10,000 steps
+- XXX music model (link to other blog post XXX)
+- Backpacks? XXX
```
Comment (Ivan-Zhou) [inline docs/Announcement-Post.md:248]: We can discuss this in the backpack sync this week.

File: docs/Announcement-Post.md [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+## Future and Conclusion
+
+This is just the beginning for Levanter. In the future, look for:
+* more models on interesting problem domains
+* scaled up versions of new architectures developed here at Stanford and elsewhere
+* new training techniques
+* larger models
+
+We are excited to continue to develop Levanter and to share our work with the community. Please join us on our journey!
+(And by the way, [we're hiring](https://crfm.stanford.edu/apply.html)!)
```
Comment (Ivan-Zhou) [inline docs/Announcement-Post.md:259]: Maybe also share us feedbacks via GH or any other public channels that we have?

**Review by Ivan-Zhou — COMMENTED** [review]
Well written announcement post 👍

File: docs/Announcement-Post.md (hunk 2 above)
Comment (dlwh) [reply docs/Announcement-Post.md:205]: thanks!

File: docs/Announcement-Post.md (hunk 4 above)
Comment (dlwh) [reply docs/Announcement-Post.md:217]: yeah, sorry laziness

File: docs/Announcement-Post.md (hunk 5 above)
Comment (dlwh) [reply docs/Announcement-Post.md:234]: thank you!

**Pull request merged**